src/coreclr/src/ToolBox/superpmi/superpmi-shared/lightweightmap.h

```diff
@@ -369,7 +369,14 @@ class LightWeightMap : public LightWeightMapBuffer
             else if (res > 0)
                 last = mid - 1; // repeat search in bottom half.
             else
+            {
+                int resItem = memcmp(&pItems[mid], &item, sizeof(_Item));
+                if (resItem != 0)
+                {
+                    LogDebug("Tried to add a new value for an existing key, the new value was ignored");
```

Comment:
I wonder how often this happens. Should it even be `LogError` or `LogWarning`? E.g., would we see this in a normal collection/replay of SPC.dll or the libraries or the coreclr tests? Are there "benign" cases that can be ignored? (And can we detect somewhere if they are benign or not?) Should the function return some indication that this has happened, so not just a `bool` return, but a 3-state return?

Reply:
It happens in 20% methods in my estimates and we later strip most of them during the clean phase. So it would be very noisy to show a warning or an error about them.

Reply:
That makes sense; thanks for the update. Yes, non-pure JIT-EE functions are a problem. It would be nice to go back and try to figure out how to avoid the "clean" phase for more cases. Hopefully we don't miss to much coverage by doing "clean".